libs/methodologies/bold/rule-processors/mass-id/src/certificate-uniqueness-check/certificate-uniqueness-check.helpers.ts

```diff
@@ -0,0 +1,58 @@
+import type {
+  BoldMethodologySlug,
+  Document,
+} from '@carrot-fndn/shared/methodologies/bold/types';
+
+import { eventHasMetadataAttribute } from '@carrot-fndn/shared/methodologies/bold/predicates';
+import { DocumentEventAttributeName } from '@carrot-fndn/shared/methodologies/bold/types';
+import { RuleOutputStatus } from '@carrot-fndn/shared/rule/types';
+import { MethodologyDocumentStatus } from '@carrot-fndn/shared/types';
+
+const { CANCELLED } = MethodologyDocumentStatus;
+
+export const hasMethodologySlugAttribute = (
+  document: Document,
+  methodologySlug: BoldMethodologySlug,
+): boolean =>
+  document.externalEvents?.some((event) =>
+    eventHasMetadataAttribute({
+      event,
+      metadataName: DocumentEventAttributeName.METHODOLOGY_SLUG,
+      metadataValues: methodologySlug,
+    }),
+  ) === true;
+
+const isMassIdAuditApproved = (document: Document): boolean =>
+  document.externalEvents?.some((event) =>
+    eventHasMetadataAttribute({
+      event,
+      metadataName: DocumentEventAttributeName.EVALUATION_RESULT,
+      metadataValues: RuleOutputStatus.APPROVED,
+    }),
+  ) === true;
+
+const isDocumentCancelled = (document: Document): boolean =>
+  document.status === CANCELLED.toString();
+
+export const hasSomeDocumentOpen = (documents: Document[]): boolean =>
+  documents.some((document) => !isDocumentCancelled(document));
+
+export const isMassIdAuditInProgress = (document: Document): boolean =>
+  !isDocumentCancelled(document) &&
+  document.externalEvents?.some(
+    (event) =>
+      !eventHasMetadataAttribute({
+        event,
+        metadataName: DocumentEventAttributeName.EVALUATION_RESULT,
+      }),
+  ) === true;
```

Comment:
I think this is less confusing:
```suggestion
  !(document.externalEvents ?? []).some(
    (event) =>
      eventHasMetadataAttribute({
        event,
        metadataName: DocumentEventAttributeName.EVALUATION_RESULT,
      }),
  );
```

Reply:
Or keep the `=== true` if the issue is with coverage

Reply:
https://github.com/carrot-foundation/methodology-rules/pull/225/commits/77bdea07c020fc22486d9f1577108e1dabd4ab99